content/en/docs/setup/install/multicluster/index.md

```diff
@@ -0,0 +1,281 @@
+
+    {{< text yaml >}}
+    mesh_id: MyMeshID 
+    clusters:
+      context-1:
+        network: network-1
+        serviceAccountReader: istio-reader-service-account        
+    {{< /text >}}
+
+1. Prepare the cluster by running the `prepcluster.sh` script created above:
+
+    {{< text bash >}}
+    $ ./prepcluster.sh
+    {{< /text >}}
+
+1. Apply the topology information:
+
+    {{< text bash >}}
+    $ istioctl x multicluster apply -f mesh-topology.yaml
+    {{< /text >}}
+
+1. Finally, make sure everything looks good:
+
+    {{< text bash >}}
+    $ istioctl x multicluster describe -f mesh-topology.yaml
+    {{< /text >}}
+
+## Setting up the second cluster
+
+The second cluster has full connectivity as the first cluster you already configured above. Setting up this
+second cluster looks a lot like what you did for the first cluster.
+
+1. Select the second cluster's context:
+
+    {{< text bash >}}
+    $ kubectl config use-context context-2
+    {{< /text >}}
+    
+1. Register the second cluster in the topology file. `mesh-topology.yaml` should now contain:
+
+    {{< text yaml >}}
+    mesh_id: MyMeshID 
+    clusters:
+      context-1:
+        network: network-1
+        serviceAccountReader: istio-reader-service-account        
+      context-2:
+        network: network-1
+        serviceAccountReader: istio-reader-service-account        
+    {{< /text >}}
+
+1. Prepare the cluster by running the `prepcluster.sh` script created above:
+
+    {{< text bash >}}
+    $ ./prepcluster.sh
+    {{< /text >}}
+
+1. Apply the topology information:
+
+    {{< text bash >}}
+    $ istioctl x multicluster apply -f mesh-topology.yaml
+    {{< /text >}}
+
+1. Finally, make sure everything looks good:
+
+    {{< text bash >}}
+    $ istioctl x multicluster describe -f mesh-topology.yaml
+    {{< /text >}}
+
+## Setting up the third cluster
+
+The third cluster doesn't have full connectivity to the first two clusters you previously configured. Setting up this
+third cluster looks a lot like what you did for the first cluster.
+
+1. Select the third cluster's context:
+
+    {{< text bash >}}
+    $ kubectl config use-context context-3
+    {{< /text >}}
+    
+1. Register the third cluster in the topology file. `mesh-topology.yaml` should now contain:
+
+    {{< text yaml >}}
+    mesh_id: MyMeshID 
+    clusters:
+      context-1:
+        network: network-1
+        serviceAccountReader: istio-reader-service-account        
+      context-2:
+        network: network-1
+        serviceAccountReader: istio-reader-service-account        
+      context-3:
+        network: network-2
+        serviceAccountReader: istio-reader-service-account        
+    {{< /text >}}
+
+1. Prepare the cluster by running the `prepcluster.sh` script created above:
+
+    {{< text bash >}}
+    $ ./prepcluster.sh
+    {{< /text >}}
+
+1. Apply the topology information:
+
+    {{< text bash >}}
+    $ istioctl x multicluster apply -f mesh-topology.yaml
```

Comment:
You need to re-generate cluster-3's manifest to include the external gateway IPs of cluster-1 and cluster-2. You also need to re-generate and re-apply cluster-1 and cluster-2's manifests to include cluster-3s.

Reply:
We should be able to specific the ingress gateway service name for each network's gateway and let Pilot discover the IP through normal service discovery (see https://github.com/istio/api/blob/6818c03d25b255739cb71b5ca4b469aa97429bd4/mesh/v1alpha1/network.proto#L80). Unfortunately, this doesn't seem to work for more than two clusters. Once that is fixed we won't need to re-generate/apply cluster config when new clusters are added.